API.php

```diff
@@ -198,6 +198,17 @@ public function getConfiguredCustomDimensions($idSite)
         return $configs;
     }
 
+    /**
+     * For convenience. Hidden to reduce API surface area.
+     * @hide
+     */
+    public function getConfiguredCustomDimensionsHavingScope($idSite, $scope)
+    {
+        $result = $this->getConfiguredCustomDimensions($idSite);
+        $result = array_filter($result, function ($row) use ($scope) { return $row['scope'] == $scope; });
```

Comment:
do we need an `array_values` around `array_filter` on top? Not sure how array_filter behaves here. If it doesn't return a properly indexed array the API response may be wrong and eg converted to an object instead of array and some other features may not work. Be good to also add a system test for it.

Reply:
Updated.